### ISSUE TITLE
Markdown doesn't automatically render on pasting content or on existing line

### ISSUE THREAD

**Issue by alterhuman** [superlistapp/super_editor#1590]
When you type the syntax first, the markdown is instantly rendered but it's not working in case of pasting text or on an existing line.

For example if I start with ## and then type `Some heading` it works but if I directly paste `## Some heading`, it doesn't work.

Similarly for lists, if start with `1.` then it works, but if I type `This is a line` and then type `1.` at the start, it doesn't work.

**Comment by matthew-carroll** [comment]
Please provide reproduction steps for your issue. Based on your description, I have no idea where you're doing these things that you're describing, or what code you're using.

**Comment by matthew-carroll** [comment]
`SuperEditor` is a tool for all editors, not just markdown editors, therefore it wouldn't be appropriate to automatically treat pasted content as markdown. However, you can implement that behavior yourself by replacing the existing paste behavior with your own, and then call the provided deserialization method that turns markdown into a `Document`.

**Comment by matthew-carroll** [comment]
I stared to explain this but then I decided that we probably make this harder than it needs to be. There will always be some work that needs to be done, but we can add some pieces that help.

When pasting Markdown text, that text needs to be deserialized into a `Document` and then the nodes in that partial Markdown text `Document` need to be inserted into the primary document. We should consider adding requests and commands to insert nodes from outside a `Document` into the primary `Document` through the `Editor`, and/or...

We might want to add requests and commands within the `super_editor_markdown` package that do the deserialization on your behalf and inserts the appropriate nodes into the primary document.

I'll look into this.
